data/refprog/reference-program/src/main/java/com/farao_community/farao/data/refprog/reference_program/ReferenceProgram.java

```diff
@@ -19,56 +16,60 @@
  */
 public class ReferenceProgram {
     private final List<ReferenceExchangeData> referenceExchangeDataList;
-    private final Set<Country> countries;
-    private final Map<Country, Double> netPositions;
+    private final Set<ReferenceProgramArea> referenceProgramAreas;
+    private final Map<ReferenceProgramArea, Double> netPositions;
 
     public ReferenceProgram(List<ReferenceExchangeData> referenceExchangeDataList) {
         this.referenceExchangeDataList = referenceExchangeDataList;
-        this.countries = new HashSet<>();
+        this.referenceProgramAreas = new HashSet<>();
         this.referenceExchangeDataList.stream().forEach(referenceExchangeData -> {
-            if (referenceExchangeData.getAreaOut() != null) {
-                countries.add(referenceExchangeData.getAreaOut());
+            referenceProgramAreas.add(referenceExchangeData.getAreaOut());
+            referenceProgramAreas.add(referenceExchangeData.getAreaIn());
+            if (referenceExchangeData.getAreaOut() != null && !referenceExchangeData.getAreaOut().isVirtualHub()) {
```

Comment:
why do you filter the referenceExchangeData for which the areaOut is virtual ?
Will it be changed in the PR on the loop-flow computation ?

Reply:
You're right, I removed some useless lines here (areaOut and areaIn were added in all cases so these checks were useless)